### PR TITLE
WT-7092 Reduce hash calls during cursor open/close

### PR DESCRIPTION
**Pull request by raviprakashgiri29**
This change is to set the computed hash value in `cursor->uri_hash`while opening a cursor and with this approach, we will reduce one hash computation during cursor close.

### CODE REVIEW

**Review by ddanderson — COMMENTED** [review]
I left a comment about `__wt_cursor_get_hash`.  Let's talk in your morning about this.

File: src/session/session_api.c [hunk 1]
```diff
@@ -531,6 +531,27 @@ __session_open_cursor_int(WT_SESSION_IMPL *session, const char *uri, WT_CURSOR *
...
+        *uri = to_dup->uri;
+        hash_value = to_dup->uri_hash;
+    } else {
+        WT_ASSERT(session, uri != NULL);
+        hash_value = __wt_hash_city64(*uri, strlen(*uri));
+    }
+
+    return (hash_value);
+}
+
```
[the remark this answers is not in the document]
Comment (raviprakashgiri29) [reply src/session/session_api.c]: @ddanderson Updated the code as per the comments.

File: src/cursor/cur_std.c [hunk 2]
```diff
@@ -756,19 +756,36 @@ __wt_cursor_cache_release(WT_SESSION_IMPL *session, WT_CURSOR *cursor, bool *rel
     return (ret);
 }
 
+/*
+ * __wt_cursor_get_hash --
+ *     Set hash value from the given uri.
```
Comment (ddanderson) [inline src/cursor/cur_std.c:761]: The description needs to change - "Get hash..." rather than "Set hash..."

File: src/session/session_api.c [hunk 3]
```diff
@@ -540,18 +540,27 @@ __wt_open_cursor(WT_SESSION_IMPL *session, const char *uri, WT_CURSOR *owner, co
...
         WT_RET_NOTFOUND_OK(ret);
     }
 
-    return (__session_open_cursor_int(session, uri, owner, NULL, cfg, cursorp));
+    ret = __session_open_cursor_int(session, uri, owner, NULL, cfg, cursorp);
+
+    if (*cursorp != NULL)
+        (*cursorp)->uri_hash = hash_value;
+
+    return (ret);
```
Comment (ddanderson) [inline src/session/session_api.c:563]: To get rid of repeating code (lines 560-561 is the same as 619-620), I would move the setting of `uri_hash` to `__session_open_cursor_int`.  That is, `__session_open_cursor_int` should take an additional argument `hash_value`, and set the result cursor's `uri_hash` there.

**Review by ddanderson — APPROVED** [review]
This is looking good.  I made a suggestion that should get rid of a repeated fragment of code.

File: src/cursor/cur_std.c (hunk 2 above)
Comment (raviprakashgiri29) [reply src/cursor/cur_std.c]: Fixed

File: src/session/session_api.c (hunk 3 above)
Comment (raviprakashgiri29) [reply src/session/session_api.c]: Fixed

**Review by sulabhM — APPROVED** [review]

**Pull request merged**